qiskit_aqua/translators/ising/exactcover.py

```diff
@@ -0,0 +1,199 @@
+        indices_has_e = np.arange(n)[cond]
+        num_has_e = len(indices_has_e)
+        Y = 1-0.5*num_has_e
+        shift += Y*Y
+
+        for i in indices_has_e:
+            for j in indices_has_e:
+                if i != j:
+                    wp = np.zeros(n)
+                    vp = np.zeros(n)
+                    vp[i] = 1
+                    vp[j] = 1
+                    pauli_list.append([0.25, Pauli(vp, wp)])
+                else:
+                    shift += 0.25
+
+        for i in indices_has_e:
+            wp = np.zeros(n)
+            vp = np.zeros(n)
+            vp[i] = 1
+            pauli_list.append([-Y, Pauli(vp, wp)])
+
+
+    return Operator(paulis=pauli_list), shift
+
+def read_numbers_from_file(filename):
+    """Read numbers from a file
+
+    Args:
+        filename (str): name of the file.
+
+    Returns:
+        numpy.ndarray: list of numbers as a numpy.ndarray.
+    """
+    numbers = []
+    with open(filename) as infile:
+        for line in infile:
+            assert(int(round(float(line)))  == float(line))
+            numbers.append(int(round(float(line))))
+    return np.array(numbers)
+
+
+
+def sample_most_likely(n, state_vector):
+    """Compute the most likely binary string from state vector.
+
+    Args:
+        n (int): number of  qubits.
+        state_vector (numpy.ndarray or dict): state vector or counts.
+
+    Returns:
+        numpy.ndarray: binary string as numpy.ndarray of ints.
+    """
+    if isinstance(state_vector, dict) or isinstance(state_vector, OrderedDict):
+        temp_vec = np.zeros(2**n)
+        total = 0
+        for i in range(2**n):
+            state = np.binary_repr(i, n)
+            count = state_vector.get(state, 0)
+            temp_vec[i] = count
+            total += count
+        state_vector = temp_vec / float(total)
+
+    k = np.argmax(np.abs(state_vector))
+    x = np.zeros(n)
+    for i in range(n):
+        x[i] = k % 2
+        k >>= 1
+    return x
+
+
+def get_solution(x):
+    """
+
+    Args:
+        x (numpy.ndarray) : binary string as numpy array.
+
+    Returns:
+        numpy.ndarray: graph solution as binary numpy array.
+    """
+    return 1 - x
+
+
+def check_solution_satisfiability(sol, list_of_subsets):
+    n = len(list_of_subsets)
+    U = []
+    for sub in list_of_subsets:
+        U.extend(sub)
+    U = np.unique(U) # U is the universe
+
+
+    U2 = []
+    selected_subsets = []
+    for i in range(n):
+        if sol[i] == 1:
+            selected_subsets.append(list_of_subsets[i])
+            U2.extend(list_of_subsets[i])
+
+    U2 = np.unique(U2)
+    if set(U) != set(U2):
+        return False
+
+    tmplen = len(selected_subsets)
+    for i in range(tmplen):
+        for j in range(i):
+            L = selected_subsets[i]
+            R = selected_subsets[j]
+
+            if set(L) & set(R): # should be empty
+                return False
+
+    return True
+
+
+
+
```

Comment:
too many spaces, do you mind checking the style with PEP8?

Reply:
done. pep8 all related files